crates/roc_app/src/wasm32.rs

```diff
@@ -1429,140 +1550,236 @@ impl PartialOrd for ConnectResult {
         Self {
-            discriminant: discriminant_ConnectResult::Connected,
-            payload: union_ConnectResult { Connected: payload },
+            discriminant: discriminant_StreamErr::BrokenPipe,
+            payload: union_StreamErr { BrokenPipe: () },
         }
     }
 
-    pub fn Error(payload: ConnectErr) -> Self {
+    pub fn ConnectionRefused() -> Self {
         Self {
-            discriminant: discriminant_ConnectResult::Error,
-            payload: union_ConnectResult {
-                Error: core::mem::ManuallyDrop::new(payload),
+            discriminant: discriminant_StreamErr::ConnectionRefused,
+            payload: union_StreamErr {
+                ConnectionRefused: (),
+            },
+        }
+    }
+
+    pub fn ConnectionReset() -> Self {
+        Self {
+            discriminant: discriminant_StreamErr::ConnectionReset,
+            payload: union_StreamErr {
+                ConnectionReset: (),
+            },
+        }
+    }
+
+    pub fn Interrupted() -> Self {
+        Self {
+            discriminant: discriminant_StreamErr::Interrupted,
+            payload: union_StreamErr { Interrupted: () },
+        }
+    }
+
+    pub fn OutOfMemory() -> Self {
+        Self {
+            discriminant: discriminant_StreamErr::OutOfMemory,
+            payload: union_StreamErr { OutOfMemory: () },
+        }
+    }
+
+    pub fn PermissionDenied() -> Self {
+        Self {
+            discriminant: discriminant_StreamErr::PermissionDenied,
+            payload: union_StreamErr {
+                PermissionDenied: (),
+            },
+        }
+    }
+
+    pub fn Unrecognized(payload: ConnectErr_Unrecognized) -> Self {
+        Self {
+            discriminant: discriminant_StreamErr::Unrecognized,
+            payload: union_StreamErr {
+                Unrecognized: core::mem::ManuallyDrop::new(payload),
             },
         }
     }
 }
 
-impl Drop for ConnectResult {
+impl Drop for StreamErr {
     fn drop(&mut self) {
         // Drop the payloads
         match self.discriminant() {
-            discriminant_ConnectResult::Connected => {}
-            discriminant_ConnectResult::Error => unsafe {
-                core::mem::ManuallyDrop::drop(&mut self.payload.Error)
+            discriminant_StreamErr::BrokenPipe => {}
+            discriminant_StreamErr::ConnectionRefused => {}
+            discriminant_StreamErr::ConnectionReset => {}
+            discriminant_StreamErr::Interrupted => {}
+            discriminant_StreamErr::OutOfMemory => {}
+            discriminant_StreamErr::PermissionDenied => {}
+            discriminant_StreamErr::Unrecognized => unsafe {
+                core::mem::ManuallyDrop::drop(&mut self.payload.Unrecognized)
             },
         }
     }
 }
 
+impl roc_std::RocRefcounted for StreamErr {
+    fn inc(&mut self) {
+        unimplemented!();
+    }
+    fn dec(&mut self) {
+        unimplemented!();
+    }
+    fn is_refcounted() -> bool {
+        true
+    }
```

Comment:
This seems problematic?

Reply:
Only matters if a list containing said type is passed between host and roc.

So if there are no effects of the sort `List (Result Stream StremErr)`, doesn't matter.

Same witht he other case pointed out.